tools/approval-service/internal/service/ttleventcache_test.go

```diff
@@ -0,0 +1,140 @@
+package service_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gravitational/shared-workflows/tools/approval-service/internal/service"
+)
+
+var (
+	eventIDForTest           = "event-id-for-test"
+	eventIDInProgressForTest = "event-id-in-progress-for-test"
+)
+
+// TestTryAddConcurrency ensures that under a burst of concurrent TryAdds for the same key
+// exactly one caller wins (returns true) and others are rejected within the TTL
+func TestTryAddConcurrency(t *testing.T) {
+	// Arrange
+	ttl := 200 * time.Millisecond
+	ec, cleanupFunc, err := service.MakeTTLEventCache(ttl)
+	if err != nil {
+		t.Fatalf("MakeTTLCache error: %v", err)
+	}
+	defer func() {
+		_ = cleanupFunc
+	}()
+
+	const goroutines = 500
+	var wg sync.WaitGroup
+
+	var wins int32
+	var mu sync.Mutex
+
+	// Act
+	for i := 0; i < goroutines; i++ {
+		wg.Go(func() {
+			if ec.TryAdd(eventIDForTest) {
```

Comment:
There is no guarantee that all of the calls to `TryAdd()` will happen within 200ms (the ttl). On a slow CI runner with lots of contention, this test could be flaky.

Could you instead use the synctest package to test this behaviour more deterministically?